c/anagrams_kata/anagrams_finder.h

```diff
@@ -0,0 +1,14 @@
+#ifndef ANAGRAMS_FINDER_H
+#define ANAGRAMS_FINDER_H
+
+typedef struct anagrams_finder_s
+{
+    char *wordlist;
+    void (*find)(struct anagrams_finder_s *_this, char *word);
```

Comment:
I am sorry, I was using the term "performance" too generously - I implied memory (resource) performance. We are talking about the same I believe, and I'd argue that this topic (loading the file as a whole / read on demand) is applicable and relevant to any language/environment; Make the file large enough and you have problems anywhere.

Not liking an aspect of a starting point makes the starting point even better for a kata, I believe. Your point about the loading itself is a great topic that can be addressed/discussed either at the beginning, or in the retrospective of a dojo. I for instance was focused only on the storage in memory itself (see #5) which is why I prefer an array over a _set for the starting point_ ; I also don't like to have an array, yet putting in a set already primes people.
Of course, it could be argued that the whole process of loading the file could be removed from the kata, to drive people to think into this aspect as well. From my experience from running several dojos, it is better for such prepared katas to have some running skeleton. At the discretion of the facilitator, they can then drive the people to focus on a certain aspect (the anagram algorithm, or the file loading behaviour, or something different).

Thank you also for the answer regarding the structure member. Interesting approach.